test/spec/services/apiServiceSpec.js

```diff
@@ -118,21 +118,58 @@ describe("APIService", function() {
       ]
     ];
     _.each(rgvs, _.spread(function(rgv, expectedAPIInfo) {
-      it('should result in ' + JSON.stringify(expectedAPIInfo) + ' when called with ' + JSON.stringify(rgv), function() {
-        var actualAPIInfo = APIService.apiInfo(rgv);
-        if(actualAPIInfo) {
-          expect(actualAPIInfo.resource).toEqual(expectedAPIInfo.resource);
-          expect(actualAPIInfo.group).toEqual(expectedAPIInfo.group);
-          expect(actualAPIInfo.version).toEqual(expectedAPIInfo.version);
-          expect(actualAPIInfo.hostPort).toEqual(expectedAPIInfo.hostPort);
-          expect(actualAPIInfo.prefix).toEqual(expectedAPIInfo.prefix);
-          expect(actualAPIInfo.namespaced).toEqual(expectedAPIInfo.namespaced);
-          expect(actualAPIInfo.verbs).toEqual(expectedAPIInfo.verbs);
-        } else {
-          expect(actualAPIInfo).toEqual(expectedAPIInfo);
-        }
-      });
+        it('should return the correct resource', function() {
+          var actualAPIInfo = APIService.apiInfo(rgv);
+          if(actualAPIInfo) {
+            expect(actualAPIInfo.resource).toEqual(expectedAPIInfo.resource);
+          }
+        });
+        it('should return the correct group', function() {
+          var actualAPIInfo = APIService.apiInfo(rgv);
+          if(actualAPIInfo) {
+            expect(actualAPIInfo.group).toEqual(expectedAPIInfo.group);
+          }
+        });
+        it('should return the correct version', function() {
+          var actualAPIInfo = APIService.apiInfo(rgv);
+          if(actualAPIInfo) {
+            expect(actualAPIInfo.version).toEqual(expectedAPIInfo.version);
+          }
+        });
+        it('should return the correct hostPort', function() {
+          var actualAPIInfo = APIService.apiInfo(rgv);
+          if(actualAPIInfo) {
+            expect(actualAPIInfo.hostPort).toEqual(expectedAPIInfo.hostPort);
+          }
+        });
+        it('should return the correct prefix', function() {
+          var actualAPIInfo = APIService.apiInfo(rgv);
+          if(actualAPIInfo) {
+            expect(actualAPIInfo.prefix).toEqual(expectedAPIInfo.prefix);
+          }
+        });
+        it('should return the correct namespaced', function() {
+          var actualAPIInfo = APIService.apiInfo(rgv);
+          if(actualAPIInfo) {
+            expect(actualAPIInfo.namespaced).toEqual(expectedAPIInfo.namespaced);
+          }
+        });
+        it('should return the correct verbs', function() {
+          var actualAPIInfo = APIService.apiInfo(rgv);
+          if(actualAPIInfo) {
+            expect(actualAPIInfo.verbs).toEqual(expectedAPIInfo.verbs);
+          }
+        });
     }));
+
+    it('should return undefined if the rgb is unknown', function() {
```

Comment:
if you are making this a separate test case, remove the case in L115-117, it won't get tested by your checks above

Reply:
Yeah, i thought that 115 just looked too much like a real resource, so the purpose of the test was obscured (given I couldn't easily diagnose).

Eliminating.